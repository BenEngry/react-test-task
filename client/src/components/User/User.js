import { useState, useEffect } from 'react';
import { useCookies } from "react-cookie";
import { GET_URL } from "../utils/consts";
import { Spinner } from "../Spinner/Spinner";

export const User = () => {
	const [error, setError] = useState(null);
    const [isLoaded, setIsLoaded] = useState(false);
    const [items, setItems] = useState();

	const [setCookie] = useCookies(["user"]);
    const [token, setToken] = useState();

    useEffect(() => {
    	fetch( GET_URL + window.location.href.split("/").reverse()[0])
        	.then( response => response.json())
			.then(
			(result) => {
                setIsLoaded(true);
                setItems(result.items[0]);
				setToken(result.token);
				setCookie("user", token, {
					path: "/"
				});
			},
			(error) => {
                setIsLoaded(true);
                setError(error);
			}
        )
    }, []);

	if (error) {
        return <div>{error.message}</div>;
    } else if (!isLoaded) {
        return <Spinner/>;
    } else {
		return (
			<a href="/">
				<div className="wrapper">
					<div className="list__item">
						<p className="list__string">Name: {items.name}</p>
						<p className="list__string">Surname: {items.surname}</p>
						<p className="list__string">Age: {items.age}</p>
						<p className="list__string">State: {items.gender}</p>
						<p>State: {items.state}</p>
					</div>
				</div>
			</a>
		);
	}
}
