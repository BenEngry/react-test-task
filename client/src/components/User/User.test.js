import renderer from "react-test-renderer";
import React from "react";
import { User } from "./User";

describe("Testing <User/>", ()=> {

    it("User have rendered correctly", ()=>{
        const user = renderer.create(<User/>).toJSON();
        expect(user).toMatchSnapshot();
    });

});
