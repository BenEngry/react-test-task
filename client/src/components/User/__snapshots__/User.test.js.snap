// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <User/> List have rendered correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="loader"
  >
    Loading...
  </div>
</div>
`;

exports[`Testing <User/> User have rendered correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="loader"
  >
    Loading...
  </div>
</div>
`;
